tests/table/test_puffin.py

```diff
@@ -71,3 +71,78 @@ def test_map_high_vals() -> None:
 
     with pytest.raises(ValueError, match="Key 4022190063 is too large, max 2147483647 to maintain compatibility with Java impl"):
         _ = _deserialize_bitmap(puffin)
+
+
+def test_puffin_round_trip() -> None:
+    # Define some deletion positions for multiple files
+    deletions1 = [10, 20, 30]
```

Comment:
Iceberg deletion vectors expect 1:1 for DV:path. The current PuffinWriter may violate the expectation. 

For instance, the following code writes 2 blobs without merging them. Do you fix this behavior as a follow-up, or the caller should provide the merged DV? 
```py
    deletions1a = [10, 20]
    deletions1b = [30]
...
    writer.add(positions=deletions1a, referenced_data_file=file1_path)
    writer.add(positions=deletions1b, referenced_data_file=file1_path)
```